src/test/java/net/snowflake/ingest/streaming/internal/StreamingIngestIT.java

```diff
@@ -93,6 +93,71 @@ public void afterAll() throws Exception {
     jdbcConnection.createStatement().execute(String.format("drop database %s", testDb));
   }
 
+  @Test
+  public void testColumnNameQuotes() throws Exception {
+    final String tableName = "T_COLUMN_WITH_QUOTES";
+    final String quotedColumn = "\"c 2\""; // user name <c 2>
+    final String doubleQuotedColumn = "\"c\"\"a\\ \"\"\\ 3\""; // user name <c"a\ "\ 3>
+    final String escapedSpacesColumn =
+        "c\\ 4"; // space is escaped with '\' in Snowflake SQL, user name <C 4>
+    jdbcConnection
+        .createStatement()
+        .execute(
+            String.format(
+                "create or replace table %s.%s.%s (c1 NUMBER(38,0), %s NUMBER(38,0), %s"
+                    + " NUMBER(38,0), %s NUMBER(38,0));",
+                testDb,
+                TEST_SCHEMA,
+                tableName,
+                quotedColumn,
+                doubleQuotedColumn,
+                escapedSpacesColumn));
+
+    OpenChannelRequest request1 =
+        OpenChannelRequest.builder("CHANNEL")
+            .setDBName(testDb)
+            .setSchemaName(TEST_SCHEMA)
+            .setTableName(tableName)
+            .setOnErrorOption(OpenChannelRequest.OnErrorOption.CONTINUE)
+            .build();
+
+    // Open a streaming ingest channel from the given client
+    SnowflakeStreamingIngestChannel channel1 = client.openChannel(request1);
+    for (int val = 0; val < 10; val++) {
+      Map<String, Object> row = new HashMap<>();
+      row.put("c1", val);
+      row.put(quotedColumn, val);
+      row.put(doubleQuotedColumn, val);
+      row.put("c 4", val);
```

Comment:
we can but the escape '\' is added because quoting is optional in UI
Java strings are always quoted, and this is what comes from server side in this case
I can add handling of escaping in this case but w/o escape it works out of the box